dev_guide/builds.adoc

```diff
@@ -1184,6 +1190,80 @@ spec:
 ----
 ====
 
+[[build-hooks]]
```

Comment:
How about we put this section after triggers? Deployments talk about triggers first, hooks later.
In general the sections are not ordered mirroring deployments, but I think in this case it makes more sense to have hooks later.